libs/wingsdk/src/platform/platform-manager.ts

```diff
@@ -99,8 +106,19 @@ export class PlatformManager {
         newInstanceOverrides.push(instance.newInstance.bind(instance));
       }
     });
+    
+    const registrar = new ParameterRegistrar("PlatformParametersRegistrar");
+    
+    registerParameterHooks.forEach((hook) => {
+      hook(registrar);
+    });
 
-    return appCall!({ ...appProps, synthHooks, newInstanceOverrides }) as App;
+    // synth the registrar before returning the app, this way
+    // the inputs are registered before the app is synthesized
+    registrar.synth();
+    
+    const app = appCall!({ ...appProps, synthHooks, newInstanceOverrides, platformParameterRegistrar: registrar }) as App;
+    return app;
```

Comment:
Want to see if I can get this registrar construct into the App tree,  ran into issues when first playing with this, and got frustrated